src/integrationtests/java/com/aws/greengrass/integrationtests/ConfigTest.java

```diff
@@ -78,6 +89,88 @@ void GIVEN_Greengrass_with_mqtt_bridge_WHEN_multiple_config_changes_consecutivel
+        // change config values
+        CountDownLatch bridgeRestarted = new CountDownLatch(1);
+
+        testContext.getKernel().getContext().addGlobalStateChangeListener((GreengrassService service, State was, State newState) -> {
+            if (service.getName().equals(MQTTBridge.SERVICE_NAME) && newState.equals(State.NEW)) {
+                bridgeRestarted.countDown();
+            }
+        });
+
+        config.lookup(BridgeConfig.KEY_SESSION_EXPIRY_INTERVAL).withValue(1);
+        config.lookup(BridgeConfig.KEY_MAXIMUM_PACKET_SIZE).withValue(1);
+        config.lookup(BridgeConfig.KEY_RECEIVE_MAXIMUM).withValue(1);
+
+        // verify that config changes take effect
+        assertTrue(bridgeRestarted.await(AWAIT_TIMEOUT_SECONDS, TimeUnit.SECONDS));
+        TimeUnit.MILLISECONDS.sleep(500);
+
+        assertEquals(1, testContext.getLocalV5Client().getSessionExpiryInterval());
+        assertEquals(1, testContext.getLocalV5Client().getMaximumPacketSize());
+        assertEquals(1, testContext.getLocalV5Client().getReceiveMaximum());
+
+        // publish large message to IoT Core and verify that it is not received due to the local client's config
+        byte[] payload = new byte[1000000];
+        Random random = new Random();
+        random.nextBytes(payload);
+
+        MqttMessage expectedMessage = MqttMessage.builder()
+                .topic("topic/toIotCore")
+                .payload(payload)
+                .userProperties(Collections.singletonList(new UserProperty("key", "val")))
+                .responseTopic("response topic")
+                .messageExpiryIntervalSeconds(1234L)
+                .payloadFormat(Publish.PayloadFormatIndicator.UTF8)
+                .contentType("contentType")
+                .build();
+
+        Pair<CompletableFuture<Void>, Consumer<Publish>> subscribeCallback
+                = asyncAssertOnConsumer(p -> assertEquals(expectedMessage, MqttMessage.fromSpoolerV5Model(p)));
+
+        testContext.getIotCoreClient().getIotMqttClient().subscribe(Subscribe.builder()
+                .topic("topic/toIotCore")
+                .callback(subscribeCallback.getRight())
+                .build());
+
+        testContext.getLocalV5Client().publish(
+                MqttMessage.builder()
+                        .topic("topic/toIotCore")
+                        .payload(payload)
+                        .userProperties(Collections.singletonList(new UserProperty("key", "val")))
+                        .responseTopic("response topic")
+                        .messageExpiryIntervalSeconds(1234L)
+                        .payloadFormat(Publish.PayloadFormatIndicator.UTF8)
+                        .contentType("contentType")
+                        .build());
+
+        // We expect this to timeout since the local client should not have been able to publish the message due to
+        // its size, so the IoT core client will have no message to retrieve
+        assertThrows(TimeoutException.class,
```

Comment:
Ah I see, I'll update it to use the local client to subscribe

Reply:
Does [this test](https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/blob/main/src/integrationtests/java/com/aws/greengrass/integrationtests/BridgeTest.java#L135) need to be updated as well? Curious since it publishes and subscribes the same way I do here, are AWS credentials somehow set for that test?

Reply:
Got it, makes sense

Reply:
and ensure that a small message is received correctly.